updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -0,0 +1,350 @@
+              # us which cannot be applied.
+              all_updated_dependencies.concat(updated_dependencies)
+              generate_dependency_files_for(lead_dependency, updated_dependencies, dependency_files)
+            else
+              dependency_files # pass on the existing files if no updates are possible
+            end
+          end
+
+          Dependabot::DependencyChange.new(
+            job: job,
+            dependencies: all_updated_dependencies,
+            updated_dependency_files: updated_files
+          )
+        end
+
+        # This method determines which dependencies must change given a target
+        # 'lead' dependency we want to update.
+        #
+        # This may return more than 1 dependency since the ecosystem-specific
+        # tooling may find collaborators which need to be updated in lock-step.
+        #
+        # This method **must** must return an Array when it errors
+        def compile_updates_for(dependency, dependency_files)
+          checker = update_checker_for(dependency, dependency_files, raise_on_ignored: raise_on_ignored?(dependency))
+
+          log_checking_for_update(dependency)
+
+          # FIXME: Grouped updates currently do not interact with ignore rules
+          # return [] if all_versions_ignored?(dependency, checker)
+
+          if checker.up_to_date?
+            log_up_to_date(dependency)
+            return []
+          end
+
+          requirements_to_unlock = requirements_to_unlock(checker)
+          log_requirements_for_update(requirements_to_unlock, checker)
+
+          if requirements_to_unlock == :update_not_possible
+            Dependabot.logger.info(
+              "No update possible for #{dependency.name} #{dependency.version}"
+            )
+            return []
+          end
+
+          updated_deps = checker.updated_dependencies(
+            requirements_to_unlock: requirements_to_unlock
+          )
+
+          if peer_dependency_should_update_instead?(checker.dependency.name, updated_deps)
+            Dependabot.logger.info(
+              "No update possible for #{dependency.name} #{dependency.version} (peer dependency can be updated)"
+            )
+            return []
+          end
+
+          filter_unrelated_and_unchanged(updated_deps, checker)
+        rescue Dependabot::InconsistentRegistryResponse => e
+          error_handler.log_error(
+            dependency: dependency,
+            error: e,
+            error_type: "inconsistent_registry_response",
+            error_detail: e.message
+          )
+          [] # return an empty set
+        rescue StandardError => e
+          error_handler.handle_dependabot_error(error: e, dependency: dependency)
+          [] # return an empty set
+        end
+
+        def filter_unrelated_and_unchanged(updated_dependencies, checker)
+          updated_dependencies.reject do |d|
+            next false if d.name == checker.dependency.name
```

Comment:
any concerns with casing?

Reply:
I think it's ok at this point as we are using `Dependabot::Dependency` objects and the name should be normalised, plus this is currently replicating [existing behaviour](https://github.com/dependabot/dependabot-core/pull/6884/files#diff-be79fd5c595a9f52f3ab443d0487b65c13d3b2b65d5b8793fb85ff91601862b5R133-R136) that hasn't been DRY'd out yet.

Case checking is something we have a few inconsistencies on though, so it's a good spot.